Sources/ComposableArchitecture/SwiftUI/Binding.swift

```diff
@@ -346,6 +364,50 @@ public struct BindingAction<Root>: Equatable {
     ) -> Bool {
       keyPath == bindingAction.keyPath
     }
+
+    #if DEBUG
+      private class Debugger<Value> {
+        let value: Value
+        let bindableActionType: Any.Type?
+        let file: StaticString
+        let line: UInt
+        var wasCalled = false
+
+        init(value: Value, bindableActionType: Any.Type?, file: StaticString, line: UInt) {
+          self.value = value
+          self.bindableActionType = bindableActionType
+          self.file = file
+          self.line = line
+        }
+
+        deinit {
+          guard self.wasCalled else {
+            let action = """
+              \(bindableActionType.map { "\($0).binding(" } ?? "\(BindingAction.self)")\
+              .set(_, \(self.value))\
+              \(bindableActionType != nil ? ")" : "")
+              """
+            runtimeWarning(
+              """
+              A binding action created at "%@:%d" never invoked its setter:
+
+                Action:
+                  %@
+
+              Enhance reducers that receive binding actions with the "Reducer.binding()" modifier \
+              to ensure the setter will mutate associated state.
```

Comment:
How do you feel about something like this?

```suggestion
              A binding action created at "%@:%d" was not properly handled:

                Action:
                  %@

              To fix this invoke the "binding()" method on your feature's reducer.
```